src/main/java/com/yahoo/bullet/rest/query/SseQueryHandler.java

```diff
@@ -0,0 +1,54 @@
+/*
+ *  Copyright 2018, Yahoo Inc.
+ *  Licensed under the terms of the Apache License, Version 2.0.
+ *  See the LICENSE file associated with the project for terms.
+ */
+package com.yahoo.bullet.rest.query;
+
+import com.yahoo.bullet.pubsub.Metadata;
+import com.yahoo.bullet.pubsub.PubSubMessage;
+import com.yahoo.bullet.rest.service.QueryService;
+import lombok.AllArgsConstructor;
+import org.springframework.web.servlet.mvc.method.annotation.SseEmitter;
+
+import java.io.IOException;
+
+/**
+ * Query handler that implements results for SSE - multiple results per query.
+ */
+@AllArgsConstructor
+public class SseQueryHandler extends QueryHandler {
+    private String queryID;
+    private SseEmitter emitter;
+    private QueryService queryService;
+
+    @Override
+    public void complete() {
+        super.complete();
+        emitter.complete();
+    }
+
+    @Override
+    public void send(PubSubMessage response) {
+        if (!isComplete()) {
+            try {
+                emitter.send(response.getContent());
+            } catch (IOException e) {
+                queryService.submitSignal(queryID, Metadata.Signal.KILL);
```

Comment:
Do you want to kill the query if one response fails for some reason or do we assume that a single failure implies future responses will fail too? In other words, do we want a retry mechanism?

Reply:
If exception thrown when sseemitter sending message, sseemitter will call comleteWithError and no more messages can be sent after that.